product_supplierinfo_discount/tests/test_product_supplierinfo_discount.py

```diff
@@ -88,14 +87,13 @@ def test_004_prepare_purchase_order_line(self):
             'state': 'confirmed',
             'warehouse_id': self.env.ref('stock.warehouse0').id,
             'move_dest_id': self.env.ref('stock.stock_location_customers').id,
-            'message_unread_counter': 0,
             'name': 'WH: Stock -> Customers MTO',
             'product_id': self.product.id,
-            'date_planned': fields.Datetime.now(),
             'rule_id': procurement_rule.id,
         }
-        procurement_order = self.env['procurement.order'].create(vals)
-        res = procurement_order._prepare_purchase_order_line(
+        # procurement_rule = self.env['procurement.rule'].create(vals)
```

Comment:
Remove this line and the full dictionary

Reply:
I can't remove vals dictionary because is used later to prepare purchase order line.

Reply:
Yeah, I saw it later. Sorry for the noise.